dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/validation/DataValidationTask.java

```diff
@@ -673,10 +673,16 @@ else if ( existingPeriodInterval > periodInterval )
         {
             grid = analyticsService.getAggregatedDataValues( params );
         }
-        catch ( PersistenceException ex )
+        catch ( PersistenceException ex ) // No data
         {
             return map;
         }
+        catch ( RuntimeException ex ) // Other error
+        {
+            log.error( DebugUtils.getStackTrace( ex ) );
+
+            return map;
```

Comment:
Could this hide real exceptions? Do we want to return the validation result if exceptions happened and we might get a partial result?

Reply:
Good questions. Without this fix, any real exceptions are not caught here, but the entire validation run aborts. The exceptions are caught at a higher level and logged in the backend, and no validation results at all are returned as if all validation rules succeeded. For example, run data quality on SL Demo and select all rules. You will never see any results because some rules hit errors in analytics.

With this fix, any real exceptions are caught here and logged in the backend (so the logging part is the same as without the fix), however no analytics results are returned into the validation formula computation, which proceeds based on datavalue table values (if any), and also proceeds to evaluate any other validation rules which may find valid results based on datavalue and/or analytics results. For example run data quality on SL Demo and select all rules. You will get results.